pythonPackages.mnist: init at 0.2.2

dmrauh opened the pull request:
###### Motivation for this change
Add python mnist utilities.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

dotlambda commented:
@GrahamcOfBorg build python2.pkgs.mnist python3.pkgs.mnist

dotlambda commented on pkgs/development/python-modules/mnist/default.nix:
```diff
@@ -0,0 +1,25 @@
+  ];
+
+  meta = with lib; {
+   description = "MNIST dataset";
```
```suggestion
   description = "Python utilities to download and parse the MNIST dataset";
```
no longDescription needed

dotlambda reviewed (changes requested):
Please add yourself to `maintainer-list.nix` in a separate commit and change the commit message of the existing one to the PR's title.

> Ran 0 tests in 0.000s

Please provide an appropriate `checkPhase` or, if there are no tests, set `doCheck = false` and add a comment with the reason.

Finally, please make the python2 build work.

dmrauh committed:
maintainers: add dmrauh

dotlambda commented on pkgs/development/python-modules/mnist/default.nix:
```diff
@@ -0,0 +1,26 @@
+  };
+
+  propagatedBuildInputs = [
+  numpy
```
```suggestion
    numpy
```

dotlambda commented on pkgs/development/python-modules/mnist/default.nix:
```diff
@@ -0,0 +1,26 @@
+
+  propagatedBuildInputs = [
+  numpy
+  ] ++ lib.optionals (isPy27) [ mock ];
```
```suggestion
  ] ++ lib.optional isPy27 mock;
```
I'm pretty sure that should only be a test dependency. You should make upstream aware of that.

dotlambda commented on pkgs/development/python-modules/mnist/default.nix:
```diff
@@ -0,0 +1,26 @@
+  numpy
+  ] ++ lib.optionals (isPy27) [ mock ];
+
+  # There exist no tests
```
There are tests: https://github.com/datapythonista/mnist/tree/mnist-0.2.2/tests. However, they're probably not distributed on PyPI.

aanderse commented:
@dmrauh are you able to address the comments above and continue with this PR?

dmrauh commented:
Please excuse my long hiatus. I hope that i have addressed all comments. The only problem i had was that some tests using sockets during check phase were failing which i had to disable in the end.

dmrauh commented:
@dotlambda @aanderse do you think my PR can be merged or is there anything else to do?

aanderse commented:
@dmrauh I don't see any issues. Thanks for following up. @dotlambda good to go from your perspective?

jonringer commented on pkgs/top-level/python-packages.nix:
```diff
@@ -770,7 +770,9 @@ in {
-  monkeyhex = callPackage ../development/python-modules/monkeyhex { };
+  mnist = callPackage ../development/python-modules/mnist { };
+
+  monkeyhex = callPackage ../development/python-modules/monkeyhex { }; 
```
I feel like monkeyhex wasn't meant to be added, please remove

dmrauh replied:
I accidentally added a space after monkeyhex, hence it was added. I fixed it now.

jonringer commented on pkgs/development/python-modules/mnist/default.nix:
```diff
@@ -0,0 +1,36 @@
+{ buildPythonPackage, fetchFromGitHub, isPy27, lib, mock, numpy, pytest
```
pytestCheckHook will bring pytest into scope
```suggestion
{ buildPythonPackage, fetchFromGitHub, isPy27, lib, mock, numpy

jonringer commented on pkgs/development/python-modules/mnist/default.nix:
```diff
@@ -0,0 +1,36 @@
+
+  propagatedBuildInputs = [ numpy ] ++ lib.optional isPy27 mock;
+
+  checkInputs = [ pytest pytestCheckHook ];
```
```suggestion
  checkInputs = [ pytestCheckHook ];
```

dmrauh replied:
Thanks for the hints! I removed pytest accordingly.

dmrauh committed:
pythonPackages.mnist: init at 0.2.2

jonringer reviewed (approved):
`nix-review` passes on NixOS
diff LGTM
commits LGTM

```
[3 built, 0.0 MiB DL]
https://github.com/NixOS/nixpkgs/pull/56518
3 package were built:
python27Packages.mnist python37Packages.mnist python38Packages.mnist
```

dmrauh commented:
Thanks to all reviewers for your help!

jonringer commented:
... it  did take 8 months... that's quite a while. what's impressive to me, is that this means we've had 17k PRs in 8 months, which is averaging like 2.1k a month O.o
